tests/core/sinks/test_format_checker.py

```diff
@@ -0,0 +1,223 @@
+"""Test the custom format validator."""
+
+from __future__ import annotations
+
+import fastjsonschema
+import pytest
+
+from singer_sdk.sinks.core import Sink
+from singer_sdk.target_base import Target
+
+
+@pytest.fixture
+def test_schema():
+    """Return a test schema."""
+
+    return {
+        "type": "object",
+        "properties": {
+            "datetime_col": {"type": "string", "format": "date-time"},
+        },
+    }
+
+
+@pytest.fixture
+def target():
+    """Return a target object."""
+
+    class CustomTarget(Target):
+        name = "test_target"
+
+    return CustomTarget()
+
+
+@pytest.fixture
+def default_sink(target, test_schema):
+    """Return a default sink object."""
+
+    class CustomSink(Sink):
+        """Custom sink class."""
+
+        def process_batch(self, context: dict) -> None:
+            pass
+
+        def process_record(self, record: dict, context: dict) -> None:
+            pass
+
+    return CustomSink(target, "test_stream", test_schema, None)
+
+
+@pytest.fixture
+def draft7_sink(target, test_schema):
+    """Return a sink object with Draft7 checks enabled."""
+
+    class CustomSink(Sink):
+        """Custom sink class."""
+
+        CHECK_RECORD_FORMATS = True
+
+        def process_batch(self, context: dict) -> None:
+            pass
+
+        def process_record(self, record: dict, context: dict) -> None:
+            pass
+
+    return CustomSink(target, "test_stream", test_schema, None)
+
+
+@pytest.fixture
+def custom_sink(target, test_schema):
+    """Return a sink object with Draft7 checks enabled and a custom datetime check."""
+
+    class CustomSink(Sink):
+        """Custom sink class."""
+
+        CHECK_RECORD_FORMATS = True
+
+        def get_validator_formats(self) -> dict:
+            """Get formats for JSON schema validator.
+
+            Returns:
+                A dictionary containing regex strings and callables
+            """
+            return {
+                "date-time": (
+                    r"^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])"
+                    r"T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?"
+                    r"(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$"
+                ),
+            }
+
+        def process_batch(self, context: dict) -> None:
+            pass
+
+        def process_record(self, record: dict, context: dict) -> None:
+            pass
+
+    return CustomSink(target, "test_stream", test_schema, None)
```

Comment:
I added this but now I am getting error `fastjsonschema.exceptions.JsonSchemaDefinitionException: Unknown format: date`.  The `date` and `time` formats are part of Draft7.  I think validators are supposed to allow formats if it doesn't know of them?

>  A JSON Schema validator will ignore any format type that it does not understand.
https://json-schema.org/understanding-json-schema/reference/string#format

Reply:
If I change `CHECK_RECORD_FORMATS = False` and found that this seems to turn off both schema and record format checks since I didn't get the error message anymore.

Reply:
Ah gotcha, how about adding `"$schema": "http://json-schema.org/draft-07/schema"` to the draft7 sink?

Reply:
I can but `fastpythonschema` defaults to draft7 when no schema is specified.  Since I haven't seen `$schema` in any of schema messages generated by SDK taps I was reluctant to add it.

> When not specified, the latest implementation is used (draft-07).

Reply:
Ah gotcha, I missed that. Disregard then 😄.